### PR TITLE
add dynamic hyperparameter scaling

### PR DESCRIPTION
**Pull request by GeorgWa**
Dynamic scaling of batch size and learning rate in NN classifier.
- Prevents FDR collapse for small libraries
- Improves speed for large libraries
Overrides static config of batchsize and lr

### CODE REVIEW

File: alphadia/fdrexperimental.py [hunk 1]
```diff
@@ -905,6 +908,42 @@ def predict_proba(self, x: np.ndarray):
         return self.network(torch.Tensor(x)).detach().numpy()
 
 
+def get_scaled_training_params(df, base_lr=0.001, max_batch=1024, min_batch=64):
```
Comment (anna-charlotte) [inline alphadia/fdrexperimental.py:911]: Is it intentional that the max batch_size is not closer to 5000 anymore, as we used it previously?

**Review by anna-charlotte — APPROVED** [review]
lgtm!

File: alphadia/fdrexperimental.py (hunk 1 above)
Comment (GeorgWa) [reply alphadia/fdrexperimental.py:911]: This kinda sneaked in here :D 
I saw improved performance going down from 5000 to 1000

File: alphadia/workflow/peptidecentric.py [hunk 2]
```diff
@@ -95,7 +95,11 @@
 ]
 
 classifier_base = fdrx.BinaryClassifierLegacyNewBatching(
-    test_size=0.001, batch_size=5000, learning_rate=0.001, epochs=10
+    test_size=0.001,
+    batch_size=5000,
```
Comment (vbrennsteiner) [inline alphadia/workflow/peptidecentric.py:99]: You mention in a previous comment that a batch size closer to 1000 is actually optimal, would it make sense to change this default?

File: tests/unit_tests/test_fdrx_base.py [hunk 3]
```diff
@@ -45,3 +47,31 @@ def test_target_decoy_fdr(mock_show):
...
+        (2_000_000, 1024, 0.001),
+        # Mid-size dataset cases
+        (500_000, 512, 0.001 * np.sqrt(512 / 1024)),  # 50% of max
+        (250_000, 256, 0.001 * np.sqrt(256 / 1024)),  # 25% of max
+        # Small dataset cases
+        (50_000, 64, 0.001 * np.sqrt(64 / 1024)),  # Should hit min batch size
+        (1_000, 64, 0.001 * np.sqrt(64 / 1024)),  # Should hit min batch size
+    ],
+)
+def test_get_scaled_training_params(n_samples, expected_batch, expected_lr):
```
Comment (vbrennsteiner) [inline tests/unit_tests/test_fdrx_base.py:66]: Same again, I would recommend 'expected_batch_size' for clarity

**Review by vbrennsteiner — APPROVED** [review]
LGTM!
